examples/dynamic/mne_var_connectivity.py

```diff
@@ -0,0 +1,215 @@
+events, event_id = mne.events_from_annotations(raw)
+
+# get sample at which seizure starts
+onset_id = event_id['onset']
+onset_idx = np.argwhere(events[:, 2] == onset_id)
+onset_sample = events[onset_idx, 0].squeeze()
+onset_sec = onset_sample / raw.info['sfreq']
+
+# remove all data after the seizure onset
+raw = raw.crop(tmin=0, tmax=onset_sec, include_tmax=False)
+
+##############################################################################
+# Create Windows of Data (Epochs) Using MNE-Python
+# ------------------------------------------------
+# We have a continuous iEEG snapshot that is about 60
+# seconds long (after cropping). We would like to estimate
+# a VAR model over a sliding window of 500 milliseconds with
+# a 250 millisecond step size.
+#
+# We can use `mne.make_fixed_length_epochs` to create an
+# Epochs data structure representing this sliding window.
+
+epochs = make_fixed_length_epochs(raw=raw, duration=0.5, overlap=0.25)
+times = epochs.times
+ch_names = epochs.ch_names
+
+print(epochs)
+
+##############################################################################
+# Compute the VAR model for all windows
+# -------------------------------------
+# Now, we are ready to compute our VAR model.
+# This will compute a VAR model for each Epoch and
+# return an EpochConnectivity data structure.
+# Each Epoch here represents the VAR model in the window
+# of data. Taken together, these represent a
+# a time-varying linear system.
+
+conn = var(data=epochs.get_data(), times=times, names=ch_names)
+
+# this returns a connectivity structure over time
+print(conn)
+
+##############################################################################
+# Evaluate this VAR model fit
+# ---------------------------
+# We can now evaluate the model fit by computing
+# the residuals of the model and visualizing it.
+# In addition, we can evaluate the covariance of the
+# residuals.
+
+predicted_data = conn.predict(epochs.get_data())
+
+# compute residuals
+residuals = epochs.get_data() - predicted_data
+
+# visualize the residuals
+fig, ax = plt.subplots()
+ax.plot(residuals.flatten(), '*')
+ax.set(
+    title='Residuals of fitted VAR model',
+    ylabel='Magnitude'
+)
+
+# compute the covariance of the residuals
+model_order = conn.attrs.get('model_order')
+t = residuals.shape[0]
+sampled_residuals = np.concatenate(
+    np.split(residuals[:, :, model_order:], t, 0),
+    axis=2
+).squeeze(0)
+rescov = np.cov(sampled_residuals)
+
+# visualize the covariance of residuals
+fig, ax = plt.subplots()
+cax = fig.add_axes([0.27, 0.8, 0.5, 0.05])
+im = ax.imshow(rescov, cmap='viridis', aspect='equal', interpolation='none')
+fig.colorbar(im, cax=cax, orientation='horizontal')
+
+##############################################################################
+# Compute one VAR model using all epochs
+# --------------------------------------
+# By setting ``model='dynamic'``, we instead treat each
+# Epoch as a sample of the same VAR model
```

Comment:
Maybe add a short motivation of when this would be done.

Reply:
Added.